lib/puppet/indirector/catalog/static_compiler.rb

```diff
@@ -91,8 +91,11 @@ def find_and_replace_metadata(request, resource, file)
   # @param resource [Puppet::Resource] The resource to add the metadata to
   # @param metadata [Puppet::FileServing::Metadata] The metadata of the given fileserver based file
   def replace_metadata(request, resource, metadata)
-    [:mode, :owner, :group].each do |param|
-      resource[param] ||= metadata.send(param)
+    file = resource.to_ral
+    if file.parameter(:source_permissions).value == :use
```

Comment:
The File type seems to behave like this (and it seems like we could save on some metadata retrievals here):

**use**
  Client requests metadata through indirection (Puppet::FileServing::Metadata.indirection.find) and copies the values to the resource.

**ignore**
    Client requests metadata through indirection (Puppet::FileServing::Metadata.indirection.find) (why?) but does nothing with it.

**use_when_creating**
    Client requests metadata from through indirection (Puppet::FileServing::Metadata.indirection.find) and copies the value to the resource if it doesn't exist.

The current behavior of the static compiler is already inconsistent because it sets mode, group, owner from metadata regardless of source_permissions.

This change makes the static compiler consistent with non-static, with the only caveat being that the catalog is no longer static when :use_when_creating is set, and the client will have to fetch metadata from the server (because we can't tell if the file exists on the client).